GVFS/GVFS.Upgrader/UpgradeOrchestrator.cs

```diff
@@ -177,6 +177,13 @@ private bool TryRunUpgrade(out Version newVersion, out string consoleError)
                 return false;
             }
 
+            if (newGVFSVersion == null)
+            {
+                newVersion = null;

```

Comment:
Why do we need to set `newVersion` to `null` here?

Similar to the comment in `TryCheckIfUpgradeAvailable` can all of `out` parameters be initialized to `null` at the start of the method and then only set as needed from that point on?  I have trouble following why some of these are being set to `null` multiple times.

Reply:
If you set an out-param at the beginning, then the compiler can't help you ensure you set it in every branching path. Even though he is setting it to `null` in two places, they are for different return conditions. We really want to ensure that it is set before each error return condition, and setting at the beginning loses that automated check.

I believe this is a case of function over style. Yes, the code would look cleaner, but we lose something in the process.

Reply:
That's a great point, and for things like error messages I can see the benefit of setting them as close to the return as possible (rather that at the start of the method).

It does look like we still need a change here though, as `newVersion` is being set to `null` at the start of this method and so we're not getting the benefits you mentioned:

```
private bool TryRunUpgrade(out Version newVersion, out string consoleError)
{
    newVersion = null;
```

Reply:
Doh! As soon as I posted my reply I realized this was about `newVersion` and not `consoleError`.

My main comment was really about consistency, if we need to make sure an `out` parameter is getting set in each branch then it shouldn't be set at the top of the method.  If we *are* initializing an `out` parameter at the top of a method there's no need to set it to that same value before returning.